### PR TITLE
[FLOC-3846] Benchmark scenario catch up

### PR DESCRIPTION
**Pull request by jongiddy**
The benchmarking load scenarios work by generating `n` requests per second in a LoopingCall.  LoopingCall can skip iterations, if a single iteration takes longer than the interval between calls.  If that happens, a `count` > 1 is passed to the next iteration.

Current code uses this count to increase the number of seconds passed in the rate measurer, but failed to increase the number of new requests generated - it should be the rate (per second) times the number of seconds passed.

This PR fixes that problem so that the number of requests generated catches up all missed intervals.

Also rename `IRequestScenarioSetup` to the simpler name `IRequest`, since it is mainly encapsulating an arbitrary request.

### CODE REVIEW

File: benchmark/scenarios/_request_load.py [hunk 1]
```diff
@@ -54,8 +54,7 @@ class RequestLoadScenario(object):
     requests at a specified rate.
 
     :ivar reactor: Reactor to use.
-    :ivar scenario_setup: provider of the interface
-        ``IRequestScenarioSetup``.
+    :ivar scenario_setup: provider of the interface ``IRequest``.
```
Comment (zubron) [inline benchmark/scenarios/_request_load.py]: I think we should rename `scenario_setup` and `scenario_setup_instance` to something that implies that they are for requests. In the test code, `request` is used, can we use a name like that here?

File: benchmark/scenarios/_request_load.py [hunk 2]
```diff
@@ -84,8 +83,7 @@ def __init__(
         ``RequestLoadScenario`` constructor.
 
         :param reactor: Reactor to use.
-        :param scenario_setup_instance: provider of the
-            ``IRequestScenarioSetup`` interface.
+        :param scenario_setup_instance: provider of the ``IRequest`` interface.
```
Comment (zubron) [inline benchmark/scenarios/_request_load.py]: Other variable name to change.

**Comment by zubron** [comment]
This looks good, thanks. I'm happy for this to be merged once the comments have been addressed. It will also need a review from @jml.

**Comment by jml** [comment]
All good. Thanks @zubron!

**Pull request merged**